src/ui/components/components.rs

```diff
@@ -307,6 +308,16 @@ impl Chat {
         self.chat_icon.set_from_file(Some(icon_path));
         // Remove old messages
         self.clear_messages();
-        // TODO: Add new messeges
+        
+        let (tx, rx) = oneshot::channel();
+        runtime().spawn(async move
+        {
+         
+            let messages = ApiEndpoints::GetMessages(channel_id, None, 50).call(None).await.unwrap();
+            println!("{:?}", messages);
+            tx.send(messages).unwrap();
+        });
+        let message = rx.blocking_recv().unwrap();
+        //TODO: ADD a way to actually load the messages
```

Comment:
Use `append_message` function in impl Chat to add messages and `clear messages` to remove old messages.